### PR TITLE
Fix Bug in `sample` Parameter in `feature_select()`: Prevent Unintended Row Removal and Dataset Modification

### LINKED ISSUE #494: Bug: `samples` parameter in `feature_select` function reduces the rows in output
### Example code with output

Example code:
```python
    # Only for Plate 6, we want to normalize to iNFixion institution and Null and WT cells 
    # to keep consistent with how the other plates are normalized (same cell line)
    if plate == "Plate_6":
        samples = "Metadata_Institution == 'iNFixion' and (Metadata_genotype == 'Null' or Metadata_genotype == 'WT')"

    print(f"Performing normalization for {plate} using samples parameter: {samples}")

    # Step 2: Normalization
    normalized_df = normalize(
        profiles=output_annotated_file,
        method="standardize",
        samples=samples,
    )
    print("Normalized dataframe shape", normalized_df.shape)

    output(
        df=normalized_df,
        output_filename=output_normalized_file,
        output_type="parquet",
    )

    # Step 3: Feature selection
    feature_select_df = feature_select(
        output_normalized_file,
        operation=feature_select_ops,
        na_cutoff=0,
        samples=samples,
    )

    print("Feature selected dataframe shape", feature_select_df.shape)

    output(
        df=feature_select_df,
        output_filename=output_feature_select_file,
        output_type="parquet",
    )
```
Output:
```
Normalized dataframe shape (7383, 2319)
Feature selected dataframe shape (1674, 1148)
```

### Issue description

When using the `samples` parameter in the `feature_select` function, with the help of @gwaybio, we noticed that the number of rows reduced significantly. This did not happen if the `samples` parameter is removed, meaning it is defaulted to "all". 

Through talking with @axiomcura, it was determined that the issue was related to this part of the code:

```python
if samples != "all":
    population_df.query(samples, inplace=True)
```

When using query in this way, it removes all the other samples/rows in the dataframe, which means the output will be incorrect. 

NOTE: This parameter is also seen in the `normalize` function but it does not have this issue.

### Expected behavior

The expected behavior would be that the `feature_select` function will return a data frame with the same number of rows as the input when the `samples` parameter is used but will only determine which columns to keep based on only the samples specified.

### Additional information

- Pycytominer v1.1.0
- Linux operating system; Pop OS!
- Running in a jupyter notebook

### PR DESCRIPTION
**Pull request by axiomcura**
# Description

Thank you for your contribution to pycytominer!
Please _succinctly_ summarize your proposed change.
What motivated you to make this change?

This pull request closes #494, which describes when using the `sample` parameter unexpectedly removes rows and modifies the original dataset. The changes aim to resolve this issue, ensuring that the `sample` parameter behaves as intended without altering the original data and removing data.

I have also created new tests to verify that the number of rows remains unchanged when applying the `samples` parameter, whether through individual operations and a combination of multiple operations.

Please also link to any relevant issues that your code is associated with.

## What is the nature of your change?

- [X] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] New and existing unit tests pass locally with my changes.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

**Comment by gwaybio** [comment]
This looks functionally good to me @axiomcura - kudos and thanks! @d33bs could you give a quick, but close look to see if there are any technical things for Erik to consider prior to merging? One thing that I could potentially see is to create a new test (rather than append to an existing one), which would be more explicitly named to test the sample parameter (e.g., `test_feature_select_samples_paramater()`), but I'm not sure how hard or important this would be.

File: pycytominer/cyto_utils/features.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def drop_outlier_features(
 
     # Subset dataframe
     if samples != "all":
-        population_df.query(samples, inplace=True)
+        population_df = population_df.query(samples)
```
Comment (d33bs) [inline pycytominer/cyto_utils/features.py:178]: Consider adding a comment here or replacing the one above this if-branch to better describe what's happening here and why. For example, the comment in `correlation_threshold` provides more information (but may also need to be updated).

File: pycytominer/operations/correlation_threshold.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def correlation_threshold(
 
     # Subset dataframe and calculate correlation matrix across subset features
     if samples != "all":
-        population_df.query(samples, inplace=True)
+        population_df = population_df.query(samples)
```
Comment (d33bs) [inline pycytominer/operations/correlation_threshold.py:51]: Consider also adding or enhancing the code comment above this if-branch to provide a description of what this is doing.

File: pycytominer/operations/get_na_columns.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def get_na_columns(population_df, features="infer", samples="all", cutoff=0.05):
     """
 
     if samples != "all":
-        population_df.query(samples, inplace=True)
+        population_df = population_df.query(samples)
```
Comment (d33bs) [inline pycytominer/operations/get_na_columns.py:36]: Similar comment here: consider adding more detail through a code comment which describes what's happening here and why.

File: pycytominer/operations/noise_removal.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def noise_removal(
     """
     # Subset dataframe
     if samples != "all":
-        population_df.query(samples, inplace=True)
+        population_df = population_df.query(samples)
```
Comment (d33bs) [inline pycytominer/operations/noise_removal.py:46]: Similar comment here: consider adding more detail through a code comment which describes what's happening here and why.

File: pycytominer/operations/variance_threshold.py [hunk 5]
```diff
@@ -51,7 +51,7 @@ def variance_threshold(
 
     # Subset dataframe
     if samples != "all":
-        population_df.query(samples, inplace=True)
+        population_df = population_df.query(samples)
```
Comment (d33bs) [inline pycytominer/operations/variance_threshold.py:54]: For each one of these modified `population_df` lines consider adding in the named parameter within the call.
```suggestion
        population_df = population_df.query(expr=samples)
```

File: tests/test_feature_select.py [hunk 6]
```diff
@@ -169,6 +169,58 @@ def test_feature_select_noise_removal():
     pd.testing.assert_frame_equal(result4b, expected_result4b)
     pd.testing.assert_frame_equal(result5b, expected_result5b)
 
+    # testing samples query leveraging metadata
```
Comment (d33bs) [inline tests/test_feature_select.py:172]: Comment applies to the modifications here within tests. I agree with @gwaybio and would suggest that this becomes a new test here to help limit the lines of code found within this single test.

File: pycytominer/operations/noise_removal.py (hunk 4 above)
Comment (d33bs) [inline pycytominer/operations/noise_removal.py:46]: Seeing how these functions which implement a `query` pattern made me wonder if there's an opportunity in future changes to abstract the functionality into a decorator or class which can help provide the capability without the repetition. I mention this as a thought towards continuous improvement and not an expectation within this PR for review.

**Review by d33bs — COMMENTED** [review]
Nice job @axiomcura! I added some comments with this review and would be interested in getting your response. Please don't hesitate to let me know if you have any questions.

Generally I also wondered:
- Have we covered the change in functionality with new tests? I noticed there was a test for `feature_select` but perhaps not the other functions which were modified. Consider adding tests for these as well to ensure the sampling edge cases are covered (to avoid surprises later).

File: pycytominer/operations/get_na_columns.py (hunk 3 above)
Comment (axiomcura) [reply pycytominer/operations/get_na_columns.py:36]: I have fixed this problem, but I've also noticed that there wasn't a check for the cutoff value. Therefore, I added a check and some tests to reflect this implementation.

**Review by d33bs — APPROVED** [review]
Thanks @axiomcura for making these changes! LGTM
